src/PhpParser/Comparing/NodeComparator.php

```diff
@@ -35,26 +35,25 @@ public function printWithoutComments(Node | array | null $node): string
      */
     public function areNodesEqual(Node | array | null $firstNode, Node | array | null $secondNode): bool
     {
-        if ($firstNode instanceof Node && $secondNode === null) {
+        if ($firstNode instanceof Node && !$secondNode instanceof Node) {
             return false;
         }
-
-        if ($secondNode instanceof Node && $firstNode === null) {
+        if (!$firstNode instanceof Node && $secondNode instanceof Node) {
             return false;
         }
 
         if (is_array($firstNode)) {
             Assert::allIsAOf($firstNode, Node::class);
 
-            if ($secondNode === null) {
+            if (!is_array($secondNode)) {
                 return false;
             }
         }
 
         if (is_array($secondNode)) {
             Assert::allIsAOf($secondNode, Node::class);
 
-            if ($firstNode === null) {
+            if (!is_array($firstNode)) {
                 return false;
             }
         }
```

Comment:
goal of the changes is that we prevent the costly AST-print-compare as often as possible